pkgs/tools/security/pass/default.nix

```diff
@@ -77,52 +96,171 @@ stdenv.mkDerivation rec {
     # dependencies (s.el) here. The user has to do this themselves.
     mkdir -p "$out/share/emacs/site-lisp"
     cp "contrib/emacs/password-store.el" "$out/share/emacs/site-lisp/"
-  '' + optionalString dmenuSupport ''
+  '' + lib.optionalString dmenuSupport ''
     cp "contrib/dmenu/passmenu" "$out/bin/"
   '';
 
-  wrapperPath = with lib; makeBinPath ([
-    coreutils
-    findutils
-    getopt
-    git
-    gnugrep
-    gnupg
-    gnused
-    tree
-    which
-    qrencode
-    procps
-  ] ++ optional stdenv.isDarwin openssl
-    ++ optional x11Support xclip
-    ++ optional waylandSupport wl-clipboard
-    ++ optionals (waylandSupport && dmenuSupport) [ ydotool dmenu-wayland ]
-    ++ optionals (x11Support && dmenuSupport) [ xdotool dmenu ]
-  );
+  solutions = {
```

Comment:
My impression is that this is *very* complex. The original file was 168 lines long which seems quite large already for a small tool such as pass. The new version contains 306 lines. I agree many added lines are just comments but these comments increase the impression of complexity: there is no reason to comment something simple in my opinion.

I would prefer an upstream patch grouping all external tool references into functions or variables. Then nixpkgs can *just* patch these lines. The result would be simpler and upstream would be better too.

Reply:
> there is no reason to comment something simple in my opinion.

As in outer, this is a **draft** of a change that uses an API that most of the maintainers/mentioned are likely not familiar with yet. The comments are for this audience, and I expect to pare them back to whatever the maintainers find helpful if the work moves forward.

> My impression is that this is _very_ complex.

This is, of course, the discussion the draft is here to start. Is a device with with all of its screws more complex than a device missing a few?